src/neo-vm/ExecutionEngine.cs

```diff
@@ -931,7 +932,72 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                                     newArray.Add(s.Clone());
                                 else
                                     newArray.Add(item);
-                            EvaluationStack.Push(new VMArray(newArray));
+                            context.EvaluationStack.Push(new VMArray(newArray));
+                        }
+                        break;
+
+                    // Stack isolation
+                    case OpCode.CALL_I:
+                        {
+                            int rvcount = context.OpReader.ReadByte();
+                            int pcount = context.OpReader.ReadByte();
+                            if (context.EvaluationStack.Count < pcount)
+                            {
+                                State |= VMState.FAULT;
+                                return;
+                            }
+                            ExecutionContext context_call = LoadScript(context.Script, rvcount);
```

Comment:
First of all, i love this change :) , but this is other point to consider

Maybe we need to define a new like this class:

```
class Script
{
public readonly byte[] Data;
public byte[] Hash;
...
}

```

Because i see a problem on optimization, now and before the change. When we make a `CALL` we copy all the current script in a new byte array (ExecutionContext) and yield it, but this is not neccessary, because we can use the same ScriptObject, without copy this values and save memory and time per any `CALL`

I think that in this line https://github.com/neo-project/neo-vm/pull/39/files#diff-c388295e4ed09df607113cde924c9fadR43 .net framework clone the byte array internally

Reply:
This happens too when we compute the hash of the script, this class would optimize this multiple calls a lot of

Reply:
No, byte array in .net is a reference type, which means it won't be cloned automatically.

Reply:
But not the hash, this will be calculated in all ExecutionContext right? maybe we can save a lot of calls to SHA256